File: Geeksforgeeks/Arrays/BinarySearch.py
class Solution:	
	def binarysearch(self, arr, n, k):
		# code here
		low = 0
		high = n - 1
		
		while low<=high:
		    mid = low + (high - low)//2
		    if arr[mid]==k:
		        return mid
		    elif arr[mid]<k:
		        low = mid + 1
		    else:
		        high = mid - 1
		
		return -1

File: CodingNinjas/Binary_Search.cpp
int binarySearch(int *input, int n, int val)
{
    //Write your code here
    int low = 0;
    int high = n - 1;
    
    while(low<=high) {
        int mid = (low + high)/2;
        
        if(input[mid] == val) {
            return mid;
        }
        else if(input[mid]<val) {
            low = mid + 1;
        }
        else if(input[mid]>val) {
            high = mid - 1;
        }
    }
    return -1;
}
